src/lavinmq/auth/user.cr

```diff
@@ -130,7 +140,20 @@ module LavinMQ
         }
       end
 
-      def can_write?(vhost, name) : Bool
+      def can_write?(vhost : String, name : String, cache : PermissionCache) : Bool
+        permission_revision = @permission_revision.lazy_get
+        if permission_revision != cache.revision
+          cache.clear
+          cache.revision = permission_revision
+        end
+
+        result = cache[{vhost, name}]?
+        return result unless result.nil?
+
+        cache[{vhost, name}] = can_write?(vhost, name)
```

Comment:
or
```suggestion
        if result = cache[{vhost, name}]?
          result
        else
          cache[{vhost, name}] = can_write?(vhost, name)
        end
```

Reply:
but hmm no, this means if we have permissions cached and you send in faulty permissions, we unnessecarily go through the regex check. we want it to skip that check if the sent in permission does not match the cached permission, since when a user changes the permission we will clear and update the cache.